src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -47,8 +52,10 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
 
         try {
             storage.saveAddressBook(model.getAddressBook());
+        } catch (AccessDeniedException e) {
+            throw new CommandException(String.format(FILE_OPS_PERMISSION_ERROR_MESSAGE, e.getMessage()), e);
         } catch (IOException ioe) {
-            throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+            throw new CommandException(String.format(FILE_OPS_ERROR_MESSAGE, ioe.getMessage()), ioe);
         }
```

Comment:
@Eclipse-Dominator What kind of problems would cause IOException to be thrown instead of AccessDeniedException ?

Reply:
@Eclipse-Dominator reminder about this question